source/sync/data-access-patterns/permissions.txt

```diff
@@ -360,6 +353,183 @@ change until the next session.
      write rules. Functions would not operate on a per-document basis.
    - You cannot store permissions information (such as "which document IDs may this user access?") in the user object.
 
+
+.. _flexible-sync-roles-json-reference:
+
+Flexible Sync Roles JSON Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A **Role** object can specify a specific collection's role or the default role
+used for any unspecified collection. You define field-level permissions with a
+**Fields** object in the Role.
+
+.. example::
+
+   The following JSON object describes a **Role** for a team admin. When granted this role, the user:
+
+   - May access any document for reading.
+   - May access a document for writing if the document's ``teamId`` field equals
+     the user's custom data ``teamId``.
+   - May write to any field of the ``address`` embedded object except
+     ``zipCode``.
+   - May write to the ``name`` field.
+   - May not read or write any other fields despite having access to the
```

Comment:
This is the only one in which you list the _reason_ for the resulting permissions ("due to the ``additional_fields``...") 
I would prefer to see this in each of the bullet points. Maybe a table?
Role Permissions | Reason
---------------------------
Read any document | "read": {}
---------------------|----------
etc

Reply:
Ah yes. Nice. But...

# Why are you yelling!

😆